storage/remote/queue_manager.go

```diff
@@ -367,6 +385,31 @@ func (t *QueueManager) Start() {
 	t.metrics.minNumShards.Set(float64(t.cfg.MinShards))
 	t.metrics.desiredNumShards.Set(float64(t.cfg.MinShards))
 
+	// Checking if the JSON file is corrupted
+	var check Checkpoints
```

Comment:
I think it would be nice to break this into a function like `loadCheckpoint()` rather than having all the logic in `Start()`.